File: scala/src/main/scala/leetcode/editor/cn/[88]合并两个有序数组.scala
package leetcode.editor.cn
object MergeSortedArray {
//leetcode submit region begin(Prohibit modification and deletion)
  object Solution {
    def merge(nums1: Array[Int], m: Int, nums2: Array[Int], n: Int): Unit = {
      var p1 = m - 1
      var p2 = n - 1
      var p = nums1.length - 1

      while p1 >= 0 && p2 >= 0 do
        if nums1(p1) > nums2(p2)
        then
          nums1(p) = nums1(p1)
          p1 -= 1
        else
          nums1(p) = nums2(p2)
          p2 -= 1

        p -= 1

      while p2 >= 0 do
        nums1(p) = nums2(p2)
        p2 -= 1
        p -= 1
    }
  }
//leetcode submit region end(Prohibit modification and deletion)

}

File: cpp/leetcodecn/[88]合并两个有序数组.cpp
#include <vector>
#include "util.h"
#include <map>
#include <string>
#include <utility>
#include <memory>

using namespace std;

//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
public:
    void merge(vector<int> &nums1, int m, vector<int> &nums2, int n) {
        auto p1 = nums1.rbegin() + n;
        auto p2 = nums2.rbegin();
        auto insert = nums1.rbegin();
        while (p1 != nums1.rend() && p2 != nums2.rend()) {
            if (*p1 > *p2) {
                *insert = *p1;
                p1++;
            } else {
                *insert = *p2;
                p2++;
            }
            insert++;
        }

        while (p2 != nums2.rend()) {
            *insert = *p2;
            insert++;
            p2++;
        }
    }
};
//leetcode submit region end(Prohibit modification and deletion)


int main(int argv, char *argc[]) {
}